scenario/state.py

```diff
@@ -523,6 +523,7 @@ def trigger(
         meta: Optional[Dict[str, Any]] = None,
         actions: Optional[Dict[str, Any]] = None,
         config: Optional[Dict[str, Any]] = None,
+        resources: Optional[Dict[Path, Path]] = None,
```

Comment:
Oh, another thing I wondered: why is this done as a `trigger` arg rather than on `State`? Isn't it essentially part of the state?

Reply:
2. I was more thinking in terms of `cp origin destination` semantics, but see point 4.
3. Good point, did this in a hurry, will add that.
4. The use case is different. Take the [grafana-agent charm](https://github.com/canonical/grafana-agent-k8s-operator/). It expects the existence of a `/src/grafana_dashboards` folder and the charm will raise an exception in __init__ if there is none. It's not in `State` but in `trigger` because it's part of the 'setup' of the simulation. We could call it an initial state, but it's got nothing to do with hook tools or pebble and therefore it does not belong in the `State` object. Also it's not meant to change: we don't want it to count as input nor output of our black-box state transition testing syntax. If our charm were packed, this data would be part of the zipped archive and would be present on the unit when juju unpacks it.
1. In light of this, to be more explicit, we could name the argument: `copy_to_charm_execution_root`, or `setup_charm_root`. Hence the `cp origin destination` intuition mentioned above.
Suggestions?

Reply:
I guess the `cp origin destination` doesn't feel like the right fit, because we're describing something, in this case file system state (Scenario == declarative) rather than copying (imperative). So to me the `{path: content}` approach makes a lot more sense. That's also similar to other Pythonic APIs like [`requests.request`](https://requests.readthedocs.io/en/latest/api/#requests.request), which does `files={name: file_obj}`.